tests/bdd/features/volume/nexus-info/test_feature.py

```diff
@@ -0,0 +1,168 @@
+def a_volume_that_has_been_published_and_unpublished(volume_ctx):
+    """a volume that has been published and unpublished."""
+    # This volume will become the old entry after re-publishing the volume
+    volume_ctx[VOLUME_CTX_KEY_OLD] = publish_volume()
+    unpublish_volume()
+
+
+@given("a volume that is not published")
+def a_volume_that_is_not_published():
+    """a volume that is not published."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert not hasattr(volume.spec, "target")
+
+
+@given("a volume that is published")
+def a_volume_that_is_published(volume_ctx):
+    """a volume that is published."""
+    volume_ctx[VOLUME_CTX_KEY] = publish_volume()
+
+
+@given("an existing volume")
+def an_existing_volume(an_existing_volume):
+    """an existing volume."""
+
+
+@when("the volume is deleted")
+def the_volume_is_deleted():
+    """the volume is deleted."""
+    ApiClient.volumes_api().del_volume(VOLUME_UUID)
+
+
+@when("the volume is published")
+def the_volume_is_published(volume_ctx):
+    """the volume is published."""
+    volume_ctx[VOLUME_CTX_KEY] = publish_volume()
+
+
+@when("the volume is re-published")
+def the_volume_is_republished(volume_ctx):
+    """the volume is re-published."""
+    volume_ctx[VOLUME_CTX_KEY] = publish_volume()
+
+
+@when("the volume is unpublished")
+def the_volume_is_unpublished():
+    """the volume is unpublished."""
+    unpublish_volume()
+
+
+@then("the nexus info structure should be present in the persistent store")
+def the_nexus_info_structure_should_be_present_in_the_persistent_store(volume_ctx):
+    """the nexus info structure should be present in the persistent store."""
+    volume = volume_ctx[VOLUME_CTX_KEY]
+    nexus_uuid = volume.state.target["uuid"]
+    assert ETCD_CLIENT.get_nexus_info(VOLUME_UUID, nexus_uuid) is not None
+
+
+@then("the nexus info structure should not be present in the persistent store")
+def the_nexus_info_structure_should_not_be_present_in_the_persistent_store(volume_ctx):
+    """the nexus info structure should not be present in the persistent store."""
+    volume = volume_ctx[VOLUME_CTX_KEY]
+    nexus_uuid = volume.state.target["uuid"]
+    assert ETCD_CLIENT.get_nexus_info(VOLUME_UUID, nexus_uuid) is None
+
+
+@then("the old nexus info structure should not be present in the persistent store")
+def the_old_nexus_info_structure_should_not_be_present_in_the_persistent_store(
+    volume_ctx,
+):
+    """the old nexus info structure should not be present in the persistent store."""
+    old_volume = volume_ctx[VOLUME_CTX_KEY_OLD]
+    nexus_uuid = old_volume.state.target["uuid"]
+    assert ETCD_CLIENT.get_nexus_info(VOLUME_UUID, nexus_uuid) is None
+
+
+@pytest.fixture
+def an_existing_volume():
+    volume = ApiClient.volumes_api().put_volume(
+        VOLUME_UUID, CreateVolumeBody(VolumePolicy(False), 1, VOLUME_SIZE)
+    )
+    yield
+    try:
+        ApiClient.volumes_api().del_volume(volume.spec.uuid)
+    except Exception:
```

Comment:
Should this be restricted to a NotFoundException?

Reply:
Yes, done.